thread_runner.py

```diff
@@ -30,7 +30,23 @@ def waitFor(self):
     def runLoopedFunction(self):
         while True:
             if self.thread_running:
-                self.output = self.func(**self.args)
+                sleep_counter = 0
+                tries_left = 3
```

Comment:
Ideally you should reset the amount of tries left after you have a successful execution, otherwise 3 different failure make the program crash, even if not consecutive.